saleor/graphql/account/mutations/account/account_update.py

```diff
@@ -91,3 +94,35 @@ def perform_mutation(cls, root, info: ResolveInfo, /, **data):
         user = cast(models.User, user)
         data["id"] = graphene.Node.to_global_id("User", user.id)
         return super().perform_mutation(root, info, **data)
+
+    @classmethod
+    @traced_atomic_transaction()
+    def save(cls, info: ResolveInfo, instance: models.User, cleaned_input):
+        default_shipping_address = cleaned_input.get(SHIPPING_ADDRESS_FIELD)
+        manager = get_plugin_manager_promise(info.context).get()
+        if default_shipping_address:
+            default_shipping_address.save()
+            instance.default_shipping_address = default_shipping_address
+        default_billing_address = cleaned_input.get(BILLING_ADDRESS_FIELD)
+        if default_billing_address:
+            default_billing_address.save()
+            instance.default_billing_address = default_billing_address
+
+        super().save(info, instance, cleaned_input)
+        if default_billing_address:
+            instance.addresses.add(default_billing_address)
+        if default_shipping_address:
+            instance.addresses.add(default_shipping_address)
+
+        instance.search_document = prepare_user_search_document_value(instance)
+        instance.save(update_fields=["search_document", "updated_at"])
+
+        cls.call_event(manager.customer_updated, instance)
+
+        if redirect_url := cleaned_input.get("redirect_url"):
+            cls.process_account_confirmation(
```

Comment:
you don't have `redirect_url` in the input, so this part will not be called in any case.